tests/error/linear_errors/if_both_unused_field.err

```diff
@@ -0,0 +1,7 @@
+Guppy compilation failed. Error in file $FILE:19
+
+17:    def foo(b: bool) -> int:
+18:        if b:
+19:            s = MyStruct(qubit())
+               ^
+GuppyError: Field `s.q` with linear type `qubit` is not used on all control-flow paths
```

Comment:
This is interesting. Instead of a linear variable in an outer context not being used in some control flow paths, there are variables created *within* certain control flow paths that are never used. I have no idea what the error should say

Reply:
Again, probably not for this PR

Reply:
Yeah this is the same problem as in https://github.com/CQCL/guppylang/pull/293#discussion_r1686164876